src/Filters/AuthRates.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace CodeIgniter\Shield\Filters;
+
+use CodeIgniter\Filters\FilterInterface;
+use CodeIgniter\HTTP\RedirectResponse;
+use CodeIgniter\HTTP\RequestInterface;
+use CodeIgniter\HTTP\Response;
+use CodeIgniter\HTTP\ResponseInterface;
+use CodeIgniter\Shield\Auth;
+
+/**
+ * Auth Rate-Limiting Filter.
+ *
+ * Provides rated limiting intended for Auth routes.
+ */
+class AuthRates implements FilterInterface
+{
+    /**
+     * Intened for use on auth form pages to restrict the number
+     * of attempts that can be generated. Restricts it to 10 attempts
+     * per minute, which is what auth0 uses.
+     *
+     * @see https://auth0.com/docs/troubleshoot/customer-support/operational-policies/rate-limit-policy/database-connections-rate-limits
+     *
+     * @param array|null $arguments
+     *
+     * @return RedirectResponse|void
+     */
+    public function before(RequestInterface $request, $arguments = null)
+    {
+        $throttler = service('throttler');
+
+        // Restrict an IP address to no more than 10 requests
+        // per minute on any auth-form pages (login, register, forgot, etc).
+        if ($throttler->check(md5($request->getIPAddress()), 10, MINUTE, 1) === false) {
+            return service('response')->setStatusCode(
```

Comment:
Friends, can you create an event here?
For when the amount of after for a bucket to completely fill. We can send a message to the user (email, SMS, etc.)
Or do you suggest a solution to this issue please?
`Events::trigger('afterAuthRatesBucketFill', $user);`

Reply:
The response already includes a message telling them how long is left until they can try again, which can be inspected on the front end and displayed to the user. I'm not convinced this has any benefit, honestly. It seems like the `failedLogin` event is already giving enough information outside of the time limit.

Reply:
@datamweb You can create your own filter based on this.

Reply:
Okay, I got it.